src/main/java/org/dependencytrack/persistence/ProjectQueryManager.java

```diff
@@ -438,7 +438,12 @@ public Project updateProject(UUID uuid, String name, String description, String
         project.setDescription(description);
         project.setVersion(version);
         project.setPurl(purl);
-        project.setActive(active);
+
+        if (project.isActive() && hasActiveChild(project) && !active){
```

Comment:
```suggestion
        if (!active && project.isActive() && hasActiveChild(project)){
```
Reihenfolge drehen -> hasActiveChild() braucht performance, die andern beiden nicht, daher das zuletzt